src/components/reusable_components/CustomGraph.js

```diff
@@ -0,0 +1,197 @@
+import React, { useState } from "react";
+import { View, TouchableOpacity, StyleSheet } from "react-native";
+import {
+  VictoryChart,
+  VictoryLine,
+  VictoryVoronoiContainer,
+  VictoryTooltip,
+} from "victory-native";
+import { Button } from "react-native-paper";
+import { THEME, LINE_GRAPH_THEME } from "../../constants/Theme";
+
+export default function CustomGraph() {
+  // Filler data until we connect to the backend
+  const mockData1 = [
+    { x: 1, y: 2 },
+    { x: 2, y: 3 },
+    { x: 3, y: 5 },
+    { x: 4, y: 4 },
+    { x: 5, y: 7 },
+  ];
+  const mockData2 = [
+    { x: 2, y: 6 },
+    { x: 3, y: 2 },
+    { x: 4, y: 9 },
+    { x: 6, y: 2 },
+    { x: 8, y: 1 },
+  ];
+
+  const mockData3 = [
+    { x: 0, y: 6 },
+    { x: 5, y: 10 },
+    { x: 6, y: 7 },
+    { x: 7, y: 9 },
+    { x: 8, y: 12 },
+  ];
+
+  const mockData4 = [
+    { x: 0, y: 6 },
+    { x: 5, y: 1 },
+    { x: 6, y: 7 },
+    { x: 7, y: 4 },
+    { x: 8, y: 10 },
+  ];
+
+  const [graphData, setGraphData] = useState(mockData1);
+  const [selectedTimeframe, setSelectedTimeframe] = useState(1);
+
+  // Update graphdata and change the selected timeframe
+  function handleTimeframeChange(timeframe) {
+    setSelectedTimeframe(timeframe);
+    switch (timeframe) {
+      case 1:
+        setGraphData(mockData1);
+        break;
+      case 2:
+        setGraphData(mockData2);
+        break;
+      case 3:
+        setGraphData(mockData3);
+        break;
+      case 4:
+        setGraphData(mockData4);
+        break;
+    }
+  }
```

Comment:
`graphData` and `handleTimeFrameChange` should be passed in as props. This graph needs to be reusable in many different parts of the application. Therefore defining these in here will be very limiting. Passing as props will allow any method to get data which means the graph can be used anywhere in the application.